### ISSUE TITLE
[LOW] Vulnerability: Firewalla SSH is vulnerable to DoS attacks

### ISSUE THREAD

**Issue by BillysCoolJob** [firewalla/firewalla#7276]
# Issue:
The version of OpenSSH server `OpenSSH_8.2p1 Ubuntu-4ubuntu0.3, OpenSSL 1.1.1f  31 Mar 2020` that is used in the Firewalla Purple (and potentially other Firewallas) as of box version 1.979 is vulnerable to DoS attacks via the chacha20-poly1305 cipher.

# Background:
Normally this issue would be HIGH because the device would be vulnerable to terrapin attacks on this version of the OpenSSH server. Specifically vulnerabilities via the following 3 CVEs:
CVE-2023-48795: General Protocol Flaw
CVE-2023-46445: Rogue Extension Negotiation Attack in AsyncSSH
CVE-2023-46446: Rogue Session Attack in AsyncSSH

However, I and Firewalla support hav confirmed that they do not use the AsyncSSH python package in the Firewalla OS, so the risk is reduced to LOW as it only leaves CVE-2023-48795: General Protocol Flaw.

# Vunerability Check
To check if the device is vulnerable, complete the following steps:
- Enable SSH on the Firewalla
- Connect and run the following command:
```
nmap --script ssh2-enum-algos -sV -p 22 localhost
```
- Check the output to see if chacha20-poly1305 is listed under the encryption algorithms.
Example Output:
```
...
|   encryption_algorithms: (6)
|       chacha20-poly1305@openssh.com
|       aes128-ctr
|       aes192-ctr
|       aes256-ctr
|       aes128-gcm@openssh.com
|       aes256-gcm@openssh.com
...
```
- Alternatively, run the following command to check the OpenSSH version:
```
ssh -V
```
Example Output:
```
pi@Firewalla:~ (Firewalla-Purple) $ ssh -V
OpenSSH_8.2p1 Ubuntu-4ubuntu0.3, OpenSSL 1.1.1f  31 Mar 2020
```

# Solutions:
## Firewalla Support Provided solution:
The Firewalla support team suggested resolving this by creating a [customized script](https://help.firewalla.com/hc/en-us/articles/360054056754-Customized-Scripting) that would disable the chacha20-poly1305 cipher in the OpenSSH configuration:
Below are tested commands to create a customized script that will disable the chacha20-poly1305 cipher in the OpenSSH configuration upon reboot or box software updates:
```
$ sudo mkdir /home/pi/.firewalla/config/post_main.d/
$ cd /home/pi/.firewalla/config/post_main.d/
$ sudo chmod +wr .
$ sudo touch /home/pi/.firewalla/config/post_main.d/chacha_fix.sh
$ sudo sh -c 'echo "#!/bin/sh" > /home/pi/.firewalla/config/post_main.d/chacha_fix.sh'
$ sudo sh -c 'echo "sudo sh -c '\''mkdir -p /etc/ssh/sshd_config.d && echo \\"Ciphers -chacha20-poly1305@openssh.com\\" > /etc/ssh/sshd_config.d/anti-dos-attack.conf'\''" >> /home/pi/.firewalla/config/post_main.d/chacha_fix.sh'
$ sudo sh -c 'echo "sudo sh -c '\''mkdir -p /etc/ssh/ssh_config.d && echo \\"Ciphers -chacha20-poly1305@openssh.com\\" > /etc/ssh/ssh_config.d/anti-dos-attack.conf'\''" >> /home/pi/.firewalla/config/post_main.d/chacha_fix.sh'
$ sudo sh -c 'echo "sudo sh -c '\''echo \"Include /etc/ssh/sshd_config.d/*.conf\" >> /etc/ssh/sshd_config'\''" >> /home/pi/.firewalla/config/post_main.d/chacha_fix.sh'
$ sudo sh -c 'echo "sudo systemctl restart sshd" >> /home/pi/.firewalla/config/post_main.d/chacha_fix.sh'
$ sudo chmod +x /home/pi/.firewalla/config/post_main.d/chacha_fix.sh
```

## Proposed Solution for the Firewalla team:
The Firewalla team should consider updating the OpenSSH server version to a non-vulnerable version, which would be any version matching or above 9.6.p1. This would eliminate the need for a customized script to disable the cipher and ensure that any Firewalla device fresh out of the box or otherwise is not vulnerable to DoS attacks. Additionally there are several other CVEs that would be resolved by upgrading the OpenSSH server version as well as the OpenSSL version which are from February (8.2p1) and March (1.1.1.f) of 2020, respectively.

**Comment by BillysCoolJob** [comment]
Yeah from the sources I have seen, there is no way to turn off SSH. Though it is exposed internally only by default. https://github.com/firewalla/firewalla/issues/602

**Comment by 0xmatt11** [comment]
@MelvinTo can we get this addressed ?

**Comment by 0xmatt11** [comment]
> I'm quite surprised to hear that firewalla, a firewall company, is using a 4 year old version of ssh in it's latest release. That would make me curious if they're **actively** keeping up with other [older] software vulnerabilities. [8.2p1 has quite a few vulns](https://www.cybersecurity-help.cz/vdb/openssh/openssh/8.2p1/)

They are **not**. See #7522

**Comment by freewil** [comment]
https://x.com/freewilmi/status/1870757656375222280

Grok Summary:

> Criticism focuses on Firewalla's failure to proactively patch vulnerabilities, with reports of unpatched issues buried in forums and ignored by official responses.

**Comment by freewil** [comment]
Firewalla has a history of ignoring maintenance and security fixes going back for at least 4 years. It's really unacceptable for a security product. There needs to be more investment in testing and security infrastructure and not just new product and feature development.

**Comment by whispy** [comment]
I agree that this is concerning, especially due to the lack of response from any contributor over a period of several months.
